[Jenkins Operator CI] - UpgradeTest#testImageChange

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

https://keycloak-jenkins.com/view/OpenShift/job/rhbk-operator-test-pipeline/1163/testReport/org.keycloak.operator.testsuite.integration/UpgradeTest/testImageChange_UpdateStrategy__3_/


```
2025-02-16 21:44:07  [ERROR] Tests run: 8, Failures: 1, Errors: 0, Skipped: 2, Time elapsed: 441.238 s <<< FAILURE! - in org.keycloak.operator.testsuite.integration.UpgradeTest
2025-02-16 21:44:07  [ERROR] org.keycloak.operator.testsuite.integration.UpgradeTest.testImageChange(UpdateStrategy)[3]  Time elapsed: 161.572 s  <<< FAILURE!
2025-02-16 21:44:07  java.lang.AssertionError: java.util.concurrent.TimeoutException
2025-02-16 21:44:07  	at org.keycloak.operator.testsuite.integration.UpgradeTest.await(UpgradeTest.java:215)
2025-02-16 21:44:07  	at org.keycloak.operator.testsuite.integration.UpgradeTest.testImageChange(UpgradeTest.java:71)
2025-02-16 21:44:07  	at java.base/java.lang.reflect.Method.invoke(Method.java:580)

```

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/.github/workflows/operator-ci.yml
+++ b/.github/workflows/operator-ci.yml
@@ -141,7 +141,6 @@ jobs:
               -Dkc.operator.keycloak.image=keycloak:${{ env.version_remote }} \
               -Dquarkus.kubernetes.env.vars.kc-operator-keycloak-image-pull-policy=Never \
               -Dtest.operator.custom.image=custom-keycloak:${{ env.version_remote }} \
-              -Dquarkus.kubernetes.env.vars.kc-operator-keycloak-update-pod-deadline-seconds=60 \
               --no-transfer-progress -Dtest.operator.deployment=remote
 
   test-olm:

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
@@ -17,15 +17,14 @@
 
 package org.keycloak.operator.testsuite.integration;
 
+import io.fabric8.kubernetes.api.model.EnvVar;
 import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.fabric8.kubernetes.api.model.MicroTime;
 import io.fabric8.kubernetes.api.model.NamespaceBuilder;
 import io.fabric8.kubernetes.api.model.Pod;
 import io.fabric8.kubernetes.api.model.Secret;
-import io.fabric8.kubernetes.api.model.Service;
 import io.fabric8.kubernetes.api.model.apps.Deployment;
 import io.fabric8.kubernetes.api.model.apps.StatefulSet;
-import io.fabric8.kubernetes.api.model.batch.v1.Job;
 import io.fabric8.kubernetes.api.model.events.v1.Event;
 import io.fabric8.kubernetes.api.model.rbac.ClusterRoleBinding;
 import io.fabric8.kubernetes.api.model.rbac.RoleBinding;
@@ -175,6 +174,9 @@ public class BaseOperatorTest implements QuarkusTestAfterEachCallback {
             ((ClusterRoleBinding)obj).getSubjects().forEach(s -> s.setNamespace(namespace));
         } else if (obj instanceof RoleBinding && "keycloak-operator-view".equals(((RoleBinding)obj).getMetadata().getName())) {
             return null; // exclude this role since it's not present in olm
+        } else if (obj instanceof Deployment) {
+            // set values useful for testing - TODO: could drive this in some way from the test/resource/application.properties
+            ((Deployment)obj).getSpec().getTemplate().getSpec().getContainers().get(0).getEnv().add(new EnvVar("KC_OPERATOR_KEYCLOAK_UPDATE_POD_DEADLINE_SECONDS", "60", null));
         }
         return obj;
     });

